pulp_rpm/app/modulemd.py

```diff
@@ -50,130 +58,121 @@ def modules_packages(modules):
+    new_module[PULP_MODULE_ATTR.PROFILES] = modulemd["data"].get("profiles")
+    new_module[PULP_MODULE_ATTR.DEPENDENCIES] = modulemd["data"].get("dependencies")
+    new_module["snippet"] = snippet
+
+    return new_module
+
+
+def create_modulemd_defaults(default, snippet):
     """
-    Get modulemd_defaults.
+    Create dict with modulemd-defaults data to can be saved to DB.
+    """
+    new_default = dict()
+    new_default[PULP_MODULEDEFAULTS_ATTR.MODULE] = default["data"].get("module")
+    new_default[PULP_MODULEDEFAULTS_ATTR.STREAM] = default["data"].get("stream", "")
+    new_default[PULP_MODULEDEFAULTS_ATTR.PROFILES] = default["data"].get("profiles")
+    new_default[PULP_MODULEDEFAULTS_ATTR.DIGEST] = hashlib.sha256(snippet.encode()).hexdigest()
 
-    Args:
-        module_index (mmdlib.ModuleIndex):
-            libmodulemd index object
+    return new_default
 
-    Returns:
-        list of modulemd_defaults as dict
 
+def create_modulemd_obsoletes(default, snippet):
     """
-    ret = list()
-    modulemd_defaults = module_index.get_default_streams().keys()
-    for module in modulemd_defaults:
-        modulemd = module_index.get_module(module)
-        defaults = modulemd.get_defaults()
-        if defaults:
-            default_stream = defaults.get_default_stream()
-            default_profile = defaults.get_default_profiles_for_stream(default_stream)
-            # create modulemd-default snippet
-            temp_index = mmdlib.ModuleIndex.new()
-            temp_index.add_defaults(defaults)
-            snippet = temp_index.dump_to_string()
-            ret.append(
-                {
-                    PULP_MODULEDEFAULTS_ATTR.MODULE: modulemd.get_module_name(),
-                    PULP_MODULEDEFAULTS_ATTR.STREAM: default_stream,
-                    PULP_MODULEDEFAULTS_ATTR.PROFILES: default_profile,
-                    PULP_MODULEDEFAULTS_ATTR.DIGEST: hashlib.sha256(snippet.encode()).hexdigest(),
-                    "snippet": snippet,
-                }
-            )
-    return ret
-
-
-def parse_obsoletes(module_names, module_index):
+    Create dict with modulemd-obsoletes data to can be saved to DB.
     """
-    Get modulemd obsoletes.
+    new_obsolete = dict()
+    new_obsolete[PULP_MODULEOBSOLETES_ATTR.MODIFIED] = datetime.strptime(
+        default["data"].get("modified"), "%Y-%m-%dT%H:%MZ"
+    )
+    new_obsolete[PULP_MODULEOBSOLETES_ATTR.MODULE] = default["data"].get("module")
+    new_obsolete[PULP_MODULEOBSOLETES_ATTR.STREAM] = default["data"].get("stream")
+    new_obsolete[PULP_MODULEOBSOLETES_ATTR.MESSAGE] = default["data"].get("message")
+    new_obsolete[PULP_MODULEOBSOLETES_ATTR.RESET] = default["data"].get("reset")
+    new_obsolete[PULP_MODULEOBSOLETES_ATTR.CONTEXT] = default["data"].get("context")
+    if default["data"].get("eol_date"):
+        new_obsolete[PULP_MODULEOBSOLETES_ATTR.EOL] = datetime.strptime(
+            default["data"].get("eol_date"), "%Y-%m-%dT%H:%MZ"
```

Comment:
Definitely pay attention to this during testing.  Perhaps a few unit tests are worthwhile for the complicated invariants because otherwise we'd need to make a lot of test fixtures.

Reply:
By the spec file, there should not be any other format that this one. Another option is to not use a date field, but not sure if good idea as `modified` is used as unique constraint.

Reply:
https://github.com/fedora-modularity/libmodulemd/blob/main/yaml_specs/modulemd_obsoletes_v1.yaml#L6-L9

Reply:
change those date fields to `TextField` as we don't edit it seems like a safer approach.

Reply:
Yes, quite possibly.  Is it represented in the YAML as a string or as a raw integer?

Reply:
It is represented as a string so keeping it as it is in yaml.
Also tested with epel and fedora modular repos (two obsoletes present).